crates/forge/tests/e2e/packages_validation.rs

```diff
@@ -0,0 +1,149 @@
+use std::{fs, process::Stdio};
+
+use super::common::runner::runner;
+use assert_fs::prelude::PathChild;
+use camino::Utf8PathBuf;
+use clap::ValueEnum;
+use forge::Template;
+use project_root::get_project_root;
+use scarb_api::ScarbCommand;
+use test_case::test_case;
+
+fn find_files_recursively(dir_path: Utf8PathBuf, file_name: &str) -> Vec<Utf8PathBuf> {
+    let mut matches = Vec::new();
+    let paths = std::fs::read_dir(dir_path).unwrap();
+
+    for path in paths {
+        let path = path.unwrap().path();
+        if path.is_dir() {
+            matches.append(&mut find_files_recursively(
+                Utf8PathBuf::from(path.to_string_lossy().to_string()),
+                file_name,
+            ));
+        } else if path.to_string_lossy().contains(file_name) {
+            matches.push(Utf8PathBuf::from(path.to_string_lossy().to_string()));
+        }
+    }
```

Comment:
I think we could use walkdir for this, no? https://docs.rs/walkdir/latest/walkdir/